server/src/main/java/org/apache/iotdb/db/metadata/MManager.java

```diff
@@ -2051,6 +2074,12 @@ IEntityMNode setUsingSchemaTemplate(IMNode node) throws MetadataException {
     // this operation may change mtree structure and node type
     // invoke mnode.setUseTemplate is invalid
     IEntityMNode entityMNode = mtree.setToEntity(node);
+
+    // to ensure alignment adapt with former node or template
+    entityMNode.setAligned(
+        node.isEntity()
+            ? node.getAsEntityMNode().isAligned()
+            : node.getUpperTemplate().isDirectAligned());
     entityMNode.setUseTemplate(true);
```

Comment:
Currently, a MNode using template is possibly not an entityMNode since the template may not has direct measurement. Thus the node type transform need check first.

Reply:
Move the field "isUsingTemplate" from EntityMNode to InternalMNode. This may be better to adapt to Tree template.

Reply:
In this method (setUsingTemplate), it will check alignment and replace it if direct measurement exists now.
To implement this, I add setUseTemplate into Interface IMNode.